tailcfg/c2ntypes.go

```diff
@@ -34,6 +34,15 @@ type C2NSSHUsernamesResponse struct {
 	Usernames []string
 }
 
+// C2NUpdateRequest is the request (from control to node) to the /update
+// handler. It tells the node what the latest version is. The node may choose
```

Comment:
// C2NUpdateRequest is the request body for POST requests to the c2n /update handler.
// It tells ...

?

Reply:
We want to support both notification-only (e.g. show an upgrade icon in the tray) and automatic update modes.

I'm torn between two API options:
1. send request in both GET (as an indication of latest available version) and POST (as a command to apply the update); this informs the client about available updates, for GUI clients to display it but not apply
2. send request only in POST; if the client has not opted in to automatic updates, it can return some new response flag `Notified` instead of `Started`

This PR does # 1, always getting the version in GET requests. But I'm open to # 2 if that's a better API design.

Reply:
So c2n is the imperative API ("do $X now").

The declarative API is MapResponse, which already has `MapResponse.ClientVersion` to just advertise an "FYI, there's a 1.x.y available now and FYI it's a security update if you care".